Currently lint-staged doesn't work well with partial commits

Reported by @PaulBone in https://github.com/devtools-html/perf.html/pull/447#issuecomment-315950229
[Here is the open issue @lint-staged](https://github.com/okonet/lint-staged/issues/62).

I also prefer the second solution.  The tools can be helpful and friendly without being "automatic".  And the user remains in full control. There's also less chance of a bug like this surprising someone in the future.

I also notice that my "yarn lint" reports a lot of the same type of error, but CircleCI does not.  They were also reported in master so I didn't worry.  Whatever solution is chosen it must allow the user to override the lint tool incase the lint tool finds false positives.  I don't know if these are false positives because I don't know what it's talking about:

    /home/paul/dev/moz/3rd_party/perf.html/src/utils/worker-factory.js
      3:1  error  Definition for rule 'for-direction' was not found  for-direction

I get 172 of these. Thanks.

--- a/package.json
+++ b/package.json
@@ -18,7 +18,6 @@
     "license-check": "devtools-license-check",
     "postcheckout": "node bin/post-checkout.js",
     "postmerge": "node bin/post-merge.js",
-    "precommit": "lint-staged",
     "preinstall": "node bin/pre-install.js",
     "prepush": "yarn test-all",
     "postrewrite": "node bin/post-rewrite.js",
@@ -105,7 +104,6 @@
     "husky": "^0.14.3",
     "jest": "^22.4.2",
     "json-loader": "^0.5.7",
-    "lint-staged": "^6.1.0",
     "local-web-server": "^2.4.0",
     "mkdirp": "^0.5.1",
     "npm-run-all": "^4.1.2",
@@ -142,11 +140,5 @@
     "resetMocks": true,
     "setupTestFrameworkScriptFile": "./src/test/setup.js",
     "verbose": true
-  },
-  "lint-staged": {
-    "*.js": [
-      "eslint --fix",
-      "git add"
-    ]
   }
 }

